segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ReadOnlySegmentContainerFactory.java

```diff
@@ -41,4 +42,9 @@ public SegmentContainer createStreamSegmentContainer(int containerId) {
                 "ReadOnly Containers can only have Id %s.", READONLY_CONTAINER_ID);
         return new ReadOnlySegmentContainer(this.storageFactory, this.executor);
     }
+
+    @Override
+    public DebugSegmentContainer createDebugStreamSegmentContainer(int containerId) {
+        return null;
```

Comment:
`throw new UnsupportedOperationException("DebugSegmentContainer not supported in ReadOnly mode.")`

Reply:
Done.